modules/accredited_representative_portal/app/sidekiq/accredited_representative_portal/replicator.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+module AccreditedRepresentativePortal
+  class Replicator
+    class ValidationError < StandardError; end
+
+    Result = Struct.new(:deleted_records, :inserted_records, keyword_init: true)
+    MODEL = UserAccountAccreditedIndividual
+    MAX_CSV_ROWS = 500
+
+    def initialize(fetcher: GithubCsvFetcher.new)
+      @fetcher = fetcher
+    end
+
+    def replicate
+      Rails.logger.info('Starting accredited individuals replication')
+
+      csv_data = @fetcher.fetch
+
+      sync!(csv_data).tap do
+        Rails.logger.info('Successfully completed accredited individuals replication')
+      end
+    rescue => e
+      Rails.logger.error(
+        "Error during accredited individuals replication: #{e.message}\n" \
+        "Backtrace: #{e.backtrace.first(5).join("\n")}"
+      )
+
+      raise
+    end
+
+    private
+
+    # rubocop:disable Rails/SkipsModelValidations
+    def sync!(csv_data)
+      validate_csv_size!(csv_data)
+
+      ActiveRecord::Base.transaction do
+        # Convert CSV data to attributes array and validate
+        new_records_attributes = csv_data.map do |row|
+          attributes = row.to_h.slice(*MODEL::SYNC_FIELDS)
```

Comment:
Can possibly define this constant within this functionality rather than on the model.